### PR TITLE
Fix: Remove nil values in should and return correct JSON

### PR DESCRIPTION
**Pull request by money**
### Jira Story

- [Fix QueryBuilder query](https://osi-cwds.atlassian.net/browse/HOT-2582)

## Description
The JSON query that was being built was not being recognized by Dora.

## Tests
- [x] I have included unit tests 
- [x] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by money** [comment]
Tested it locally.  It worked.  Though @wmitchell-cambria search for street alone does not open the autocompleter.  The results are being returned.

File: app/repositories/query_builder.rb [hunk 1]
```diff
@@ -97,7 +100,7 @@ def fields
   end
 
   def client_only
-    { match: { 'legacy_descriptor.legacy_table_name': 'CLIENT_T' } }
+    { match: { 'legacy_descriptor.legacy_table_name' => 'CLIENT_T' } }
```
Comment (wmitchell-cambria) [inline app/repositories/query_builder.rb]: Does this also fix is_client_only to properly take effect again? We this afternoon that both Snapshot and Hotline were only displaying Clients. (Hotline should display all people.)

**Comment by wmitchell-cambria** [comment]
That UI bug you mentioned is interesting. I'll have to look into that one.

**Comment by money** [comment]
@ShravanParande have you tested locally to verify is_client_only feature is broken?

**Comment by ShravanParande** [comment]
> @ShravanParande have you tested locally to verify is_client_only feature is broken?

Yes , in both hotline and snapshot search function returning clients only.

**Comment by wmitchell-cambria** [comment]
Awesome teamwork everybody.

**Comment by wmitchell-cambria** [comment]
Just double checked the is_client_only as well, and it all looks good:

* With no environment variable set, Snapshot is clients only and Hotline is all types.
* With `CLIENT_ONLY_SEARCH: "true"` in docker-compose, both are clients only.
* With `CLIENT_ONLY_SEARCH: "false"` in docker-compose, Snapshot is clients only and Hotline is all types. (Same as not specifying it.)

This is all as expected.

**Comment by money** [comment]
Thank you everyone!

**Pull request merged**